src/command-handler.ts

```diff
@@ -0,0 +1,146 @@
+import { Interaction, TextBasedChannel } from 'discord.js';
+import { deleteTrackedPlayer, insertTrackedPlayer, updateTrackingChannel } from './pg-storage';
+import { BOSSES } from 'osrs-json-hiscores';
+import { getBossName } from './boss-utility';
+import { fetchHiScores } from './hiscores';
+import { updatePlayer, replyUpdateMessage } from './util';
+import { CLUES_NO_ALL, SKILLS_NO_OVERALL, CONSTANTS } from './constants';
+import state from './instances/state';
+import logger from './instances/logger';
+
+const INVALID_TEXT_CHANNEL = 'err/invalid-text-channel';
+const UNAUTHORIZED_USER = 'err/unauthorized-user';
+
+class CommandHandler {
+    static getInteractionGuildId(interaction: Interaction) {
+        if (typeof interaction.guildId !== 'string') {
+            throw new Error(INVALID_TEXT_CHANNEL);
+        }
+        return interaction.guildId;
+    }
+
+    static assertIsAdmin(interaction: Interaction) {
+        if (!state.isAdmin(interaction.user.id)) {
+            throw new Error(UNAUTHORIZED_USER);
```

Comment:
You can also limit the default permissions of a command so it only shows up for Administrators:
- https://discord.js.org/#/docs/discord.js/main/class/ApplicationCommandManager?scrollTo=create
- https://discord.js.org/#/docs/discord.js/main/typedef/ApplicationCommandData

Reply:
Yeah I was just sort of porting functionality but it is kinda beat to leave that unfinished since it's one of the best parts about slash commands, I'll add it in.